frontend/src/scenes/session-recordings/player/snapshot-processing/process-all-snapshots.ts

```diff
@@ -35,7 +43,11 @@ export function processAllSnapshots(
         return []
     }
 
-    const seenHashes: Set<string> = new Set()
+    if (hashId !== sessionRecordingId) {
+        hashId = sessionRecordingId
+        seenHashes.clear()
+    }
```

Comment:
this is a little optimisation that we weren't storing this hash between subsequent calls
and it's more effective stored between calls
so, we keep it globally and clear it if the recording id ever changes